src/sourmash/command_sketch.py

```diff
@@ -283,3 +295,304 @@ def translate(args):
+                if p.dna: assert is_dna
+                sig = SourmashSignature.from_params(p)
+                sigs.append(sig)
+
+            input_is_protein = not is_dna
+
+            # read sequence records & sketch
+            notify(f'... reading sequences from {filename}')
+            for n, record in enumerate(screed_iter):
+                if n % 10000 == 0:
+                    if n:
+                        notify('\r...{} {}', filename, n, end='')
+
+                try:
+                    add_seq(sigs, record.sequence, input_is_protein,
+                            check_sequence)
+                except ValueError as exc:
+                    error(f"ERROR when reading from '{filename}' - ")
+                    error(str(exc))
+                    sys.exit(-1)
+
+            notify('...{} {} sequences', filename, n, end='')
+
+            set_sig_name(sigs, filename, name)
+            for sig in sigs:
+                save_sigs.add(sig)
+
+            notify(f'calculated {len(sigs)} signatures for {n+1} sequences in {filename}')
+
+
+    save_sigs.close()
+    notify(f"saved {len(save_sigs)} signature(s) to '{save_sigs.location}'. Note: signature license is CC0.")
+
+
+def _output_csv_info(filename, sigs_to_build):
+    "output information about what signatures to build, in CSV format"
+    # @CTB try/test actually using this info to build sigs :)
+    output_n = 0
+    with sourmash_args.FileOutputCSV(filename) as csv_fp:
+        w = csv.DictWriter(csv_fp, fieldnames=['filename', 'sketchtype',
+                                               'output_index', 'name',
+                                               'param_strs'])
+        w.writeheader()
+
+        output_n = 0
+        for (name, filename), param_objs in sigs_to_build.items():
+            param_strs = []
+
+            # should all be the same!
+            if param_objs[0].dna:
+                assert all( ( p.dna for p in param_objs ) )
+                sketchtype = "dna"
+            else:
+                assert not any( ( p.dna for p in param_objs ) )
+                sketchtype = "protein"
+
+            for p in param_objs:
+                param_strs.append(p.to_param_str())
+
+            row = dict(filename=filename, sketchtype=sketchtype,
+                       param_strs="-p " + " -p ".join(param_strs),
+                       name=name, output_index=output_n)
+
+            w.writerow(row)
+
+            output_n += 1
+
+
+def fromfile(args):
+    if args.license != 'CC0':
+        error('error: sourmash only supports CC0-licensed signatures. sorry!')
+        sys.exit(-1)
+
+    if args.output_signatures and os.path.exists(args.output_signatures):
+        if not args.force_output_already_exists:
+            error(f"** ERROR: output location '{args.output_signatures}' already exists!")
+            error(f"** Not overwriting/appending.")
+            error(f"** Use --force-output-already-exists if you want to overwrite/append.")
```

Comment:
is there a method to append new sigs while not overwriting existing? Just output to new location and cat them together?

Reply:
depends on output, actually - zip files will append, directories will add, .sig.gz will overwrite. I'll create an issue about that.

Reply:
https://github.com/sourmash-bio/sourmash/issues/1921